plugins/offers_invreq_hook.c

```diff
@@ -304,6 +304,27 @@ static struct command_result *found_best_peer(struct command *cmd,
 							 cast_const2(const struct tlv_encrypted_data_tlv **, etlvs),
 							 ids);
 
+		/* If they tell us to use scidd for first point, grab
+		 * a channel from node (must exist, it's public) */
+		if (dev_invoice_bpath_scid) {
+			struct gossmap *gossmap = get_gossmap(cmd->plugin);
+			struct node_id best_nodeid;
+			const struct gossmap_node *n;
+			const struct gossmap_chan *c;
+			struct short_channel_id_dir scidd;
+
+			node_id_from_pubkey(&best_nodeid, &best->id);
+			n = gossmap_find_node(gossmap, &best_nodeid);
+			c = gossmap_nth_chan(gossmap, n, 0, &scidd.dir);
```

Comment:
When/if we use a sciddir outside of testing we'll have to check for dying channels too.

Reply:
This actually works: gossmap includes dying channels (I just checked).